{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "talented-class",
   "metadata": {},
   "source": [
    "# Getting the final GTF"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "enabling-baker",
   "metadata": {},
   "source": [
    "After expanding the GTF with StringTie and MicroExonator, we further process the obtained GTF to:\n",
    "\n",
    "1.- Make sure that all new transcripts are correctly assinged to genes\n",
    "2.- All genes has at least 3 exons \n",
    "3.- Collapse begining and ends of genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "third-blake",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "stylish-recall",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_gene = dict()\n",
    "gene_exons = defaultdict(set)\n",
    "transcript_exons = defaultdict(list)\n",
    "\n",
    "\n",
    "with open(\"/lustre/scratch117/cellgen/team218/gp7/Joe/MicroExonator/Report/out.high_quality.gtf\") as file:\n",
    "    \n",
    "    for row in csv.reader(file, delimiter = '\\t'):\n",
    "\n",
    "        if row[0][0]!=\"#\":\n",
    "\n",
    "            chrom = row[0]\n",
    "            source = row[1]\n",
    "            strand =  row[6]\n",
    "            feature = row[2]\n",
    "            start = row[3]\n",
    "            end = row[4]\n",
    "\n",
    "            tags = row[8].strip(\";\").split(\"; \")\n",
    "\n",
    "            tag_dict = dict()\n",
    "\n",
    "            for t in tags:\n",
    "                tag = t.strip(\" \")\n",
    "                if len(tag.split(' '))==2:\n",
    "                    field, value = tag.split(' ')\n",
    "                    value = value.strip('\"')\n",
    "                    tag_dict[field] = value\n",
    "\n",
    "            gene_id = tag_dict[\"gene_id\"]\n",
    " \n",
    "            if feature==\"exon\":\n",
    "                gene_exons[gene_id].add((chrom, strand, start, end))\n",
    "            \n",
    "                transcript_exons[tag_dict[\"transcript_id\"]].append((chrom, strand, start, end))\n",
    "                \n",
    "                if \"ENSMUS\" in gene_id:\n",
    "                        exon_gene[(chrom, strand, start, end)] = gene_id\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fifty-packet",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_re_annotation = dict()\n",
    "\n",
    "with open(\"/lustre/scratch117/cellgen/team218/gp7/Joe/MicroExonator/Report/out.high_quality.gtf\") as file:\n",
    "    \n",
    "    for row in csv.reader(file, delimiter = '\\t'):\n",
    "\n",
    "        if row[0][0]!=\"#\":\n",
    "\n",
    "            chrom = row[0]\n",
    "            source = row[1]\n",
    "            strand =  row[6]\n",
    "            feature = row[2]\n",
    "            start = row[3]\n",
    "            end = row[4]\n",
    "\n",
    "            tags = row[8].strip(\";\").split(\"; \")\n",
    "\n",
    "            tag_dict = dict()\n",
    "\n",
    "            for t in tags:\n",
    "                tag = t.strip(\" \")\n",
    "                if len(tag.split(' '))==2:\n",
    "                    field, value = tag.split(' ')\n",
    "                    value = value.strip('\"')\n",
    "                    tag_dict[field] = value\n",
    "\n",
    "            gene_id = tag_dict[\"gene_id\"]\n",
    " \n",
    "            if \"ENSMUS\" not in gene_id:\n",
    "        \n",
    "                if feature==\"exon\":\n",
    "                \n",
    "                    exon = (chrom, strand, start, end)\n",
    "                    \n",
    "                    if exon in exon_gene:\n",
    "                \n",
    "                        new_gene_id = exon_gene[exon]\n",
    "                    \n",
    "                        gene_re_annotation[gene_id] = new_gene_id\n",
    "                \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "shared-coordinator",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_t_starts = defaultdict(list)\n",
    "gene_t_ends = defaultdict(list)\n",
    "\n",
    "with open(\"/lustre/scratch117/cellgen/team218/gp7/Joe/MicroExonator/Report/out.high_quality.gtf\") as file:\n",
    "    \n",
    "    for row in csv.reader(file, delimiter = '\\t'):\n",
    "\n",
    "        if row[0][0]!=\"#\":\n",
    "\n",
    "            chrom = row[0]\n",
    "            source = row[1]\n",
    "            strand =  row[6]\n",
    "            feature = row[2]\n",
    "            start = row[3]\n",
    "            end = row[4]\n",
    "\n",
    "            tags = row[8].strip(\";\").split(\"; \")\n",
    "\n",
    "            tag_dict = dict()\n",
    "\n",
    "            for t in tags:\n",
    "                tag = t.strip(\" \")\n",
    "                if len(tag.split(' '))==2:\n",
    "                    field, value = tag.split(' ')\n",
    "                    value = value.strip('\"')\n",
    "                    tag_dict[field] = value\n",
    "\n",
    "            gene_id = tag_dict[\"gene_id\"]\n",
    "\n",
    "            \n",
    "            \n",
    "            if gene_id in gene_re_annotation:\n",
    "                gene_id = gene_re_annotation[gene_id]\n",
    "            \n",
    "                        \n",
    "            if feature==\"transcript\":\n",
    "                \n",
    "                gene_t_starts[gene_id].append(int(start))\n",
    "                gene_t_ends[gene_id].append(int(end))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "significant-kingdom",
   "metadata": {},
   "outputs": [],
   "source": [
    "window_t_starts = []\n",
    "window_t_ends = []\n",
    "\n",
    "gene_round_starts = dict()\n",
    "gene_round_ends = dict()\n",
    "\n",
    "for gene, starts in gene_t_starts.items():\n",
    "    \n",
    "    start_bins = defaultdict(list)\n",
    "    round_starts = dict()\n",
    "    \n",
    "    for s in starts:\n",
    "        start_round = str(s)[:-2]\n",
    "        start_bins[start_round].append(s)\n",
    "        \n",
    "    for start_bin, start_list in start_bins.items():\n",
    "        round_starts[start_bin] =  max(set(start_list), key = start_list.count)\n",
    "        \n",
    "    gene_round_starts[gene]  = round_starts\n",
    "    \n",
    "        \n",
    "for gene, ends in gene_t_ends.items():\n",
    "    \n",
    "    end_bins = defaultdict(list)\n",
    "    round_ends = dict()\n",
    "    \n",
    "    for e in ends:\n",
    "        end_round = str(e)[:-2]\n",
    "        end_bins[end_round].append(e)\n",
    "        \n",
    "    for end_bin, end_list in end_bins.items():\n",
    "        round_ends[end_bin] =  max(set(end_list), key = end_list.count)\n",
    "        \n",
    "    gene_round_ends[gene]  = round_ends\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "middle-listening",
   "metadata": {},
   "outputs": [],
   "source": [
    "for gene, starts in gene_t_starts.items():\n",
    "    print(gene, sorted(starts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cleared-default",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "transcript_first_exon = dict()\n",
    "transcript_last_exon = dict()\n",
    "\n",
    "for transcript, exons in transcript_exons.items():\n",
    "    transcript_first_exon[transcript] = exons[0]\n",
    "    transcript_last_exon[transcript] = exons[-1]\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "particular-detector",
   "metadata": {},
   "outputs": [],
   "source": [
    "out = open(\"/lustre/scratch117/cellgen/team218/gp7/Joe/MicroExonator/Report/out.high_quality.calibrated.gtf\", \"w\") \n",
    "\n",
    "writer = csv.writer(out, delimiter=\"\\t\", quotechar='', quoting=csv.QUOTE_NONE, escapechar='\\\\')\n",
    "\n",
    "\n",
    "with open(\"/lustre/scratch117/cellgen/team218/gp7/Joe/MicroExonator/Report/out.high_quality.gtf\") as file:\n",
    "    \n",
    "    for row in csv.reader(file, delimiter = '\\t'):\n",
    "\n",
    "        if row[0][0]!=\"#\":\n",
    "\n",
    "            chrom = row[0]\n",
    "            source = row[1]\n",
    "            strand =  row[6]\n",
    "            feature = row[2]\n",
    "            start = row[3]\n",
    "            end = row[4]\n",
    "\n",
    "            tags = row[8].strip(\";\").split(\"; \")\n",
    "\n",
    "            tag_dict = dict()\n",
    "\n",
    "            for t in tags:\n",
    "                tag = t.strip(\" \")\n",
    "                if len(tag.split(' '))==2:\n",
    "                    field, value = tag.split(' ')\n",
    "                    value = value.strip('\"')\n",
    "                    tag_dict[field] = value\n",
    "\n",
    "            gene_id = tag_dict[\"gene_id\"]\n",
    "            \n",
    "            if gene_id in gene_re_annotation:\n",
    "                gene_id = gene_re_annotation[gene_id]\n",
    "                \n",
    "            tag_dict[\"gene_id\"] = gene_id\n",
    "\n",
    "            nexons = len(gene_exons[tag_dict[\"gene_id\"]])\n",
    "   \n",
    "            if nexons>=3:\n",
    "        \n",
    "        \n",
    "                if feature==\"exon\":\n",
    "                \n",
    "                    transcript_id = tag_dict[\"transcript_id\"]\n",
    "                    exon = (chrom, strand, start, end)\n",
    "                    \n",
    "                    \n",
    "                \n",
    "                    if gene_id in gene_round_starts and start[:-2] in gene_round_starts[gene_id] and exon==transcript_first_exon[transcript_id]:\n",
    "\n",
    "                        start = gene_round_starts[gene_id][start[:-2]]\n",
    "\n",
    "                    if gene_id in gene_round_ends and end[:-2] in gene_round_ends[gene_id] and exon==transcript_last_exon[transcript_id]:\n",
    "\n",
    "                        end = gene_round_ends[gene_id][end[:-2]]\n",
    "\n",
    "                    \n",
    "                    if int(start)>int(end):\n",
    "                        \n",
    "                        start = exon[-2]\n",
    "                        end = exon[-1]\n",
    "                        \n",
    "                    row[3] = start\n",
    "                    row[4] = end\n",
    "                        \n",
    "                new_tags_list = []\n",
    "\n",
    "                for field, value in tag_dict.items():\n",
    "\n",
    "                    pair = field + ' \"' + value + '\";'\n",
    "                    new_tags_list.append(pair) \n",
    "\n",
    "                new_tags = \" \".join(new_tags_list)\n",
    "                new_line = row[:-1] + [new_tags]\n",
    "                \n",
    "                #print(new_line)\n",
    "\n",
    "                #rint(new_line)\n",
    "\n",
    "                writer.writerow(new_line)\n",
    "                    #print(transcript_id)\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "standing-heater",
   "metadata": {},
   "outputs": [],
   "source": [
    "[1,2,3,1,3,2,1].count(10)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  },
  "toc-showcode": true
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
